Framework/Algorithms/src/SampleCorrections/MayersSampleCorrectionStrategy.cpp

```diff
@@ -74,25 +74,23 @@ namespace Algorithms {
  * Constructor
  * @param params Defines the required parameters for the correction
  * @param inputHist A histogram containing the TOF values corresponding
- * to the values to be corrected. The number of tof values must match
- * number of signal/error or be 1 greater
+ * to the values to be corrected.
  */
 MayersSampleCorrectionStrategy::MayersSampleCorrectionStrategy(
     MayersSampleCorrectionStrategy::Parameters params,
     const Mantid::HistogramData::Histogram &inputHist)
-    : m_pars(params), m_tof(inputHist.x().rawData()),
-      m_sigin(inputHist.y().rawData()), m_errin(inputHist.e().rawData()),
-      m_histogram(m_tof.size() == m_sigin.size() + 1),
-      m_muRrange(calculateMuRange()), m_rng(new MersenneTwister(1)) {
-
-  // Sanity check
-  assert(m_sigin.size() == m_tof.size() || m_sigin.size() == m_tof.size() - 1);
-  assert(m_errin.size() == m_tof.size() || m_sigin.size() == m_tof.size() - 1);
-
-  if (!(m_tof.front() < m_tof.back())) {
-    throw std::invalid_argument(
-        "TOF values are expected to be monotonically increasing");
-  }
+    : m_pars(params), m_histogram(inputHist),
+      m_histoYSize(inputHist.y().size()), m_muRrange(calculateMuRange()),
+      m_rng(new MersenneTwister(1)) {
+
+  double lastVal = m_histogram.x().front();
+  for (const auto &val : m_histogram.x())
+    if (val < lastVal) {
+      throw std::invalid_argument(
+          "TOF values are expected to be monotonically increasing");
+    } else {
+      lastVal = val;
+    }
```

Comment:
I'd either leave this sanity check as it was, or fix it properly. As it is now you are still not testing what the error message claims (i.e., monotonically increasing values).
The plan is to have such a validation as part of the `BinEdges` class one day, so I'd go for the easy option and just leave things as they were.